cmd/ctr/commands/tasks/tasks_unix.go

```diff
@@ -72,18 +72,20 @@ func NewTask(ctx gocontext.Context, client *containerd.Client, container contain
 	stdinC := &stdinCloser{
 		stdin: os.Stdin,
 	}
-	stdio := cio.NewCreator(append([]cio.Opt{cio.WithStreams(stdinC, os.Stdout, os.Stderr)}, ioOpts...)...)
+	var ioCreator cio.Creator
+	if con != nil {
```

Comment:
suggest moving this if then block to the line before NewTask if ioCreator has not been set... thus removing all unnecessary NewCreator calls..?

Reply:
> suggest moving this if then block to the line before NewTask if ioCreator has not been set... thus removing all unnecessary NewCreator calls..?

@mikebrow adopt your suggestion, PTAL. I have to cleanup some code at the same time because windows didn't take care of `StdinCloser` and `logURI`.

Reply:
I finally choose removing unnecessary NewCreator in NewTask because it won't change too much.